.github/workflows/build.yaml

```diff
@@ -0,0 +1,68 @@
+name: Build ADI br2-external
+
+on:
+  push:
+  pull_request:
+
+env:
+  BR2_DL_DIR: $HOME/.buildroot-dl
+  CC: gcc-15
+  CXX: g++-15
+
+jobs:
+  build:
+    runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+      matrix:
+        defconfig:
+          - adi_sc598_ezkit_defconfig
+
+    steps:
+      - name: Checkout br2-external
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Prepare Buildroot download dir
+        run: mkdir -p "$BR2_DL_DIR"
+
+      - name: Cache Buildroot downloads
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.BR2_DL_DIR }}
+          key: ${{ runner.os }}-buildroot-dl-${{ hashFiles('configs/**/*.fragment', 'configs/**/*.defconfig') }}
+          restore-keys: |
+            ${{ runner.os }}-buildroot-dl-
+
+      - name: Build image for ${{ matrix.defconfig }}
+        run: |
+          cd buildroot
+
+          echo ">>> Selecting defconfig: ${{ matrix.defconfig }}"
+
+          make BR2_EXTERNAL="${GITHUB_WORKSPACE}" ${{ matrix.defconfig }}
+
+          if [ -f "${GITHUB_WORKSPACE}/configs/buildroot.fragment" ] || \
+             [ -f "${GITHUB_WORKSPACE}/configs/debug.fragment" ]; then
```

Comment:
I wouldn't conditionalize this. If the files don't exist then the build should fail. This adds complexity and room for error.